mount/mount_darwin.go

```diff
@@ -0,0 +1,54 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package mount
+
+import (
+	"github.com/containerd/containerd/log"
+	"github.com/pkg/errors"
+)
+
+var (
+	// ErrNotImplementOnUnix is returned for methods that are not implemented
+	ErrNotImplementOnUnix = errors.New("not implemented under darwin")
+)
+
+// Mount to the provided target path.
+//
+// Use symlink instead of union mount on darwin
+func (m *Mount) Mount(target string) error {
+	log.L.Debugf("mount: %s, src %s", target, m.Source)
+	if m.Type != "darwin" {
```

Comment:
I'm not sure we can reach consensus on adding this pseudo filesystem type.

I think it's better to exec thirdparty `fmt.Sprintf("mount_%s", m.Type)` binary.

Reply:
> I'm not sure we can reach consensus on adding this pseudo filesystem type.

I think it's not yet.  So, this is a proposal.

Though it's a bit different one, windows snapshotter has a mount type 'windows-layer', so I was thinking it's not the first one in containerd.

> I think it's better to exec thirdparty `fmt.Sprintf("mount_%s", m.Type)` binary.

Sorry, I didn't well understand this;  Do you suggest to not use `containerd/log` package and use `fmt` instead ?

Reply:
> Sorry, I didn't well understand this;
> To accept unknown mount type ?

`mount_darwin.go` should try to execute `fmt.Sprintf("mount_%s", m.Type)` binary when the binary is present.
This design will support arbitrary thirdparty mount types.

Reply:
Thanks, now I see your point.

I'm okay to add the code, but do you really think there is a value to support various mount types without any current users ?